### PR TITLE
Add error response to be JSON

### LINKED ISSUE #323: Fleet Server doesn't response with formatted error body
Elastic Agent expects the Fleet Server to respond with a JSON response when an error occurs. This allows the Elastic Agent to output an understandable error to a user when enrolling or in other cases.

The Fleet Server should respond with the following matching struct:

```
type errResp struct {
	StatusCode int    `json:"statusCode"`
	Error      string `json:"error"`
	Message    string `json:"message"`
}
```

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

Any time the API returns an error to an HTTP request it should be JSON so it can be decoded by the Elastic Agent.

## Why is it important?

So proper and easy to understand error messages are reported by the Elastic Agent.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- ~~[ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## Related issues

- Closes #323

### CODE REVIEW

**Review by scunningham — CHANGES_REQUESTED** [review]
You are leaking raw strings to an untrusted client.  You should not provide that much information to the client.  What is the reasoning here?

**Comment by blakerouse** [comment]
@scunningham You talking about the `err.Error()`. It would be nice to give a reason, at the moment there is no information in the client.

Let's say it's a version mismatch, its unknown currently what the issue. We need to provide something.

**Comment by scunningham** [comment]
It is generally bad practice to leak out the underlying raw errors on the interface, particularly to clients that should be considered untrusted.  It would be fine to translate well known errors into documented error strings, but since  the error could come out of any subsystem, this implementation could expose specifics.

A cleaner strategy would be to log verbose errors in the fleet server, translate known useful errors to pre-baked user facing error strings, and those that are not useful should be generic errors.  Customer, if have access, can refer to the fleet server log, which should contain metadata about the transaction (src port, ip, agent_id if available, etc).

**Comment by blakerouse** [comment]
@scunningham How about now? Updated it to force the message text.

File: cmd/fleet/metrics.go [hunk 1]
```diff
@@ -101,33 +101,48 @@ func init() {
...
 
 	incFail := true
 
 	var code int
 	var errStr string
+	var msgStr string
 	switch err {
 	case ErrAgentNotFound:
 		errStr = "AgentNotFound"
+		msgStr = "agent could not be cound"
```
Comment (ruflin) [inline cmd/fleet/metrics.go:115]:
```suggestion
		msgStr = "agent could not be found"
```

**Review by scunningham — APPROVED** [review]
thanks!

**Pull request merged**